src/file_handle_cache.erl

```diff
@@ -581,6 +583,43 @@ info_keys() -> ?INFO_KEYS.
 info() -> info(?INFO_KEYS).
 info(Items) -> gen_server2:call(?SERVER, {info, Items}, infinity).
 
+clear_read_cache() ->
+    gen_server2:cast(?SERVER, clear_read_cache),
+    clear_vhost_read_cache(rabbit_vhost:list()).
+
+clear_vhost_read_cache([VHost | Rest]) ->
+    clear_queue_read_cache(rabbit_amqqueue:list(VHost)),
+    clear_vhost_read_cache(Rest);
+clear_vhost_read_cache([]) ->
+    ok.
```

Comment:
Just a pet peeve, but can you make the empty list case the first function clause? It seems to be the style favoured in this file as well: https://github.com/rabbitmq/rabbitmq-server/blob/rabbitmq-server-196/src/file_handle_cache.erl#L763

Same comment for the rest of the changes